logos_setup.py

```diff
@@ -0,0 +1,801 @@
+                pipe.write(line)
+                print(line.decode(), end="")
+
+        WINETRICKS_STATUS = proc.wait()
+        ZENITY_RETURN = proc.poll()
+
+        os.remove(pipe_winetricks)
+
+        # NOTE: sometimes the process finishes before the wait command, giving the error code 127
+        if ZENITY_RETURN == 0 or ZENITY_RETURN == 127:
+            if WINETRICKS_STATUS != 0:
+                subprocess.call([WINESERVER_EXE, "-k"])
+                logos_error("Winetricks Install ERROR: The installation was cancelled because of sub-job failure!\n * winetricks " + " ".join(args) + "\n  - WINETRICKS_STATUS: " + str(WINETRICKS_STATUS), "")
+        else:
+            subprocess.call([WINESERVER_EXE, "-k"])
+            logos_error("The installation was cancelled!\n * ZENITY_RETURN: " + str(ZENITY_RETURN), "")
+    elif DIALOG == "kdialog":
+        no_diag_msg("kdialog not implemented.")
+    else:
+        no_diag_msg("No dialog tool found.")
+
+    if VERBOSE:
+        print("winetricks", *args, "DONE!")
+
+    heavy_wineserver_wait()
+
+def winetricks_dll_install(*args):
+    WINETRICKSBIN = os.environ.get('WINETRICKSBIN')
+    VERBOSE = os.getenv('VERBOSE')
+    if VERBOSE:
+        print("winetricks", *args)
+    #logos_continue_question("Now the script will install the DLL " + " ".join(args) + ". This may take a while. There will not be any GUI feedback for this. Continue?", "The installation was cancelled!", "")
+    subprocess.call([WINETRICKSBIN, *args])
+    if VERBOSE:
+        print("winetricks", *args, "DONE!")
+    heavy_wineserver_wait()
+
+def getPremadeWineBottle():
+    WINE64_BOTTLE_TARGZ_URL = os.environ.get('WINE64_BOTTLE_TARGZ_URL')
+    WINE64_BOTTLE_TARGZ_NAME = os.environ.get('WINE64_BOTTLE_TARGZ_NAME')
+    WORKDIR = os.environ.get('WORKDIR')
+    APPDIR = os.environ.get('APPDIR')
+    VERBOSE = os.getenv('VERBOSE')
+    if VERBOSE:
+        print("Installing pre-made wineBottle 64bits…")
+    logos_reuse_download(WINE64_BOTTLE_TARGZ_URL, WINE64_BOTTLE_TARGZ_NAME, WORKDIR)
+    subprocess.call(["tar", "xzf", os.path.join(WORKDIR, WINE64_BOTTLE_TARGZ_NAME), "-C", APPDIR])
+    logos_progress("Extracting…", "Extracting: " + WINE64_BOTTLE_TARGZ_NAME + "\ninto: " + APPDIR)
+
+## END WINE BOTTLE AND WINETRICKS FUNCTIONS
+## BEGIN LOGOS INSTALL FUNCTIONS 
+def installFonts():
+    if os.environ.get('WINETRICKS_UNATTENDED') is None:
+        if os.environ.get('SKIP_FONTS') is None:
+            winetricks_install('-q', 'corefonts')
+            winetricks_install('-q', 'tahoma')
+    else:
+        if os.environ.get('SKIP_FONTS') is None:
+            winetricks_install('corefonts')
+            winetricks_install('tahoma')
+    winetricks_install('-q', 'settings', 'fontsmooth=rgb')
+
+def installD3DCompiler():
+    if os.environ.get('WINETRICKS_UNATTENDED') is None:
+        winetricks_dll_install('-q', 'd3dcompiler_47')
+    else:
+        winetricks_dll_install('d3dcompiler_47')
```

Comment:
We may want to move these to wine.py

Reply:
I did this in [048345b](https://github.com/thw26/LogosLinuxInstaller/pull/7/commits/048345b37218e916e427d984f070c5b0913ff53a), but I left `getPremadeWineBottle` in `logos_setup.py` for now. Maybe it would make sense to put it in `utils.py`, though? Actually, since it uses `logos_reuse_download`, which is defined here in `logos_setup`, that would start a cascade of functions needing to be moved to avoid circular imports. Maybe it's better left here.